lib/models/blueprint.js

```diff
@@ -849,29 +849,80 @@ Blueprint.prototype._locals = function(options) {
   var moduleName = options.entity && options.entity.name || packageName;
   var sanitizedModuleName = moduleName.replace(/\//g, '-');
 
-  return new Promise(function(resolve) {
-    resolve(this.locals(options));
-  }.bind(this)).then(function (customLocals) {
-    var fileMapVariables = this._generateFileMapVariables(moduleName, customLocals, options);
-    var fileMap = this.generateFileMap(fileMapVariables);
-    var standardLocals = {
-      dasherizedPackageName: stringUtils.dasherize(packageName),
-      classifiedPackageName: stringUtils.classify(packageName),
-      dasherizedModuleName: stringUtils.dasherize(moduleName),
-      classifiedModuleName: stringUtils.classify(sanitizedModuleName),
-      camelizedModuleName: stringUtils.camelize(sanitizedModuleName),
-      decamelizedModuleName: stringUtils.decamelize(sanitizedModuleName),
-      fileMap: fileMap,
-      hasPathToken: this.hasPathToken,
-      targetFiles: options.targetFiles,
-      rawArgs: options.rawArgs
-    };
-
-    return merge({}, standardLocals, customLocals);
+  return this._processPrompts(options).then(function (prompts) {
+    return new Promise(function(resolve) {
+      resolve(this.locals(prompts));
+    }.bind(this)).then(function (customLocals) {
+      var fileMapVariables = this._generateFileMapVariables(moduleName, customLocals, options);
+      var fileMap = this.generateFileMap(fileMapVariables);
+      var standardLocals = {
+        dasherizedPackageName: stringUtils.dasherize(packageName),
+        classifiedPackageName: stringUtils.classify(packageName),
+        dasherizedModuleName: stringUtils.dasherize(moduleName),
+        classifiedModuleName: stringUtils.classify(sanitizedModuleName),
+        camelizedModuleName: stringUtils.camelize(sanitizedModuleName),
+        decamelizedModuleName: stringUtils.decamelize(sanitizedModuleName),
+        fileMap: fileMap,
+        hasPathToken: this.hasPathToken,
+        targetFiles: options.targetFiles,
+        rawArgs: options.rawArgs
+      };
+
+      return merge({}, standardLocals, customLocals);
+    }.bind(this));
   }.bind(this));
 };
 
 /**
+  @private
+  @method _processPrompts
+  @param {Object} options
+  @return {Promise}
+*/
+Blueprint.prototype._processPrompts = function (options) {
+  if (this.ui.ci || options.skipPrompts) {
```

Comment:
what I meant was I prefer explicitly using `ember g foo bar --skip-prompts` on TravisCI etc. instead of implying the option automatically when `CI=true`